api/lib/controllers/auth/auth.js

```diff
@@ -222,49 +235,47 @@ exports.getResetToken = async (ctx) => {
 };
 
 exports.resetPassword = async (ctx) => {
-  const { token, password } = ctx.request.body;
+  const { password } = ctx.request.body;
+  const { username } = ctx.state.user;
+
+  const user = await usersService.findUnique({ where: { username } });
+  if (!user) {
+    ctx.throw(404, ctx.$t('errors.auth.noUserFound'));
+    return;
+  }
 
   let decoded;
+  // TODO create middleware to add the content of token in state in param token
   try {
-    decoded = jwt.verify(token, secret);
+    decoded = jwt.verify(ctx.request.headers.authorization.split('Bearer ')[1], secret);
   } catch (err) {
     ctx.throw(400, ctx.$t('errors.password.invalidToken'));
     return;
   }
```

Comment:
Can be omited by using requireJwt and requireUser middlewares, however requireJwt should be modified to expose the original token data (as of now, ctx.state.user is overriden by requireUser)